domain/application/state/application.go

```diff
@@ -1245,15 +1235,27 @@ func (st *State) insertCloudServiceAddresses(
 	}
 
 	insertAddressStmt, err := sqlair.Prepare(`
-INSERT INTO ip_address (*)
-VALUES ($ipAddress.*);
+INSERT INTO ip_address (uuid, net_node_uuid, device_uuid, address_value, subnet_uuid, type_id, config_type_id, origin_id, scope_id)
```

Comment:
If you do a single query to retrieve the k8s subnets up front, you can look it up for each address, and this becomes a one-liner without a join.